x-pack/plugins/task_manager/server/polling_lifecycle.test.ts

```diff
@@ -0,0 +1,116 @@
+import sinon from 'sinon';
+import { of } from 'rxjs';
+
+import { TaskPollingLifecycle, claimAvailableTasks } from './polling_lifecycle';
+import { loggingSystemMock } from '../../../../src/core/server/mocks';
+import { createInitialMiddleware } from './lib/middleware';
+import { TaskTypeDictionary } from './task_type_dictionary';
+import { taskStoreMock } from './task_store.mock';
+
+describe('TaskPollingLifecycle', () => {
+  let clock: sinon.SinonFakeTimers;
+
+  const taskManagerLogger = loggingSystemMock.create().get();
+  const mockTaskStore = taskStoreMock.create({});
+  const taskManagerOpts = {
+    config: {
+      enabled: true,
+      max_workers: 10,
+      index: 'foo',
+      max_attempts: 9,
+      poll_interval: 6000000,
+      max_poll_inactivity_cycles: 10,
+      request_capacity: 1000,
+    },
+    taskStore: mockTaskStore,
+    logger: taskManagerLogger,
+    definitions: new TaskTypeDictionary(taskManagerLogger),
+    middleware: createInitialMiddleware(),
+    maxWorkersConfiguration$: of(100),
+    pollIntervalConfiguration$: of(100),
+  };
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    taskManagerOpts.definitions = new TaskTypeDictionary(taskManagerLogger);
+  });
+
+  afterEach(() => clock.restore());
+
+  describe('start', () => {
+    test('begins poilling once start is called', () => {
+      const taskManager = new TaskPollingLifecycle(taskManagerOpts);
+
+      clock.tick(150);
+      expect(mockTaskStore.claimAvailableTasks).not.toHaveBeenCalled();
+
+      taskManager.start();
+
+      clock.tick(150);
+      expect(mockTaskStore.claimAvailableTasks).toHaveBeenCalled();
+    });
+  });
+
+  describe('claimAvailableTasks', () => {
+    test('should claim Available Tasks when there are available workers', () => {
+      const logger = loggingSystemMock.create().get();
+      const claim = jest.fn(() => Promise.resolve({ docs: [], claimedTasks: 0 }));
+
+      const availableWorkers = 1;
+
+      claimAvailableTasks([], claim, availableWorkers, logger);
+
+      expect(claim).toHaveBeenCalledTimes(1);
+    });
+
+    test('should not claim Available Tasks when there are no available workers', () => {
+      const logger = loggingSystemMock.create().get();
+      const claim = jest.fn(() => Promise.resolve({ docs: [], claimedTasks: 0 }));
+
+      const availableWorkers = 0;
+
+      claimAvailableTasks([], claim, availableWorkers, logger);
+
+      expect(claim).not.toHaveBeenCalled();
+    });
+
+    /**
+     * This handles the case in which Elasticsearch has had inline script disabled.
+     * This is achieved by setting the `script.allowed_types` flag on Elasticsearch to `none`
+     */
+    test('handles failure due to inline scripts being disabled', () => {
+      const logger = loggingSystemMock.create().get();
+      const claim = jest.fn(() => {
+        throw Object.assign(new Error(), {
+          msg: '[illegal_argument_exception] cannot execute [inline] scripts',
```

Comment:
I'm torn on this - I tend to bias towards "realistic" data as we could introduce a bug due to the input not aligning with reality.... mock data not matching reality is a very common cause of bugs.

That said, since we're using `identifyEsError` and that file has the full mocks in its code the represent reality - I decided in this case we can be flexible on this, so reduced this to just the response.